src/Observable.ts

```diff
@@ -37,6 +37,10 @@ import {ErrorObservable} from './observable/throw';
 import {AjaxCreationMethod} from './observable/dom/ajax';
 import {WebSocketSubject} from './observable/dom/webSocket';
 
+export type ObservableOrPromise<T> = Observable<T> | Promise<T>;
```

Comment:
Yup, make sense to me too. Only reason I thought of separating module is case of import without `Observable`.